File: Sudoku/code/python/Solver.py

from Square import Square
from Board import Board
from Backtracking_Forward_Checking import Backtracking_Forward_Checking
from timeit import default_timer

def Input(filename):
	def readfile(filename,delim=' '):
		with open(filename, 'r') as f:
			for line in f:
				yield list(line[:-1].split(delim))

	readline=readfile(filename)
	sudoku_board=[]
	for line in readline:
		if len(line)!=1:
			l=[]
			for t in line:
				l.append(Square(0,0,eval(t)))
			sudoku_board.append(l)

	sudoku = Board(sudoku_board)
	return sudoku


if __name__=="__main__":
	start=default_timer()
	filename='Easy.txt'
        #filename='Hard.txt'
        #filename='Medium.txt'
        #filename='Medium_1.txt'
        #filename='80_hard.txt'
	try:
		Main_Board=Input(filename)
	except ValueError:
		pass
	print Main_Board

	b_f = Backtracking_Forward_Checking(Main_Board)
	b_f.Initial_Forward_Checking()
	print "after"
	b_f.BackTrack();
	stop=default_timer()
	print Main_Board
	print "Time elapsed : ",stop-start

File: Sudoku/code/java/src/Solver.java
import java.io.File;
import java.io.IOException;
import java.util.Scanner;

public class Solver {

    public static Sudoku_Board Main_Board;
    
    public static void main(String[] args) {
        
    	long system_time = System.nanoTime();
        
          String filename = "Easy.txt";
         //String filename = "Hard.txt";
        // String filename = "Medium.txt";
        //String filename = "Medium_1.txt";
        
        
            try {
                    Main_Board = input(filename);
                    
            } catch (Exception e) {
                    
                e.printStackTrace();
                    
            }
		
        System.out.println(Main_Board.toString());

        Backtracking_Forward_Checking b_f = new Backtracking_Forward_Checking(Main_Board);
        
        b_f.Initial_Forward_Checking(); // Reduce the domain for pre_assigned values

        b_f.BackTrack();
        
        System.out.println(Main_Board.toString());
        System.out.printf("Time elapsed : %.2fms%n", (System.nanoTime() - system_time)/1000000.0);
  
    }
    
    
    public static Sudoku_Board input(String filename) throws IOException {

    	Scanner in = new Scanner(new File(filename));
			
        int dim = in.nextInt();	

        Square [][] sudoku_board = new Square[dim][dim];

        for (int i = 0; i < dim; i++) {
            
            for (int j = 0; j < dim; j++) {

                int val = in.nextInt();
                
                sudoku_board[i][j] = new Square(i,j,val);
                
            }
        }
			
            in.close();
            
            Sudoku_Board sudoku = new Sudoku_Board(sudoku_board);
            
            return sudoku;
            
	}
    
}
